pulser/simulation/simulation.py

```diff
@@ -640,44 +640,51 @@ def make_xy_term(masked: bool = False) -> qutip.Qobj:
             The units are given so that the coefficient
             includes a 1/hbar factor.
             """
-            # Check if the total number of unmasked qubits is less than 2
+            # Calculate the total number of good, unmasked qubits
             effective_size = (
-                self._size - len(self._seq._slm_mask_targets)*int(masked)
+                self._size
+                - len(self._seq._slm_mask_targets) * int(masked)
+                - sum(self._bad_atoms.values())
```

Comment:
What about when a bad atom is also a masked one? Aren't you subtracting it twice?

Reply:
Correct, I'll fix it

Reply:
Now it should be fixed. I've added a couple of unit test to check if it's working properly, just to be sure.
The check inside `make_xy_term` happens only if the masked term is being created, because for the unmasked case the check was already done outside, right?

Reply:
You mean the check for `effective_size`?

Reply:
Yes that one

Reply:
That is correct, this is enough